students/kegan/session07/test_html_render.py

```diff
@@ -0,0 +1,208 @@
+"""
+Kathryn Egan
+
+Suite of tests for components of html_render.py
+"""
+from html_render import *
+
+
+def test_element_init():
+    element = Element()
+    assert element.content == []
+    element = Element('content')
+    assert element.content == ['content']
+    element = Element('')
+    assert element.content == ['']
+
+
+def test_append():
+    element = Element()
+    element.append('content')
+    element.append('more content')
+    assert element.content == ['content', 'more content']
+
+
+# def test_render_no_indent():
+#     html = Html()
+#     html.append(Body('body content'))
+#     html.append(P())
+#     temp = 'html_test.html'
+#     with open(temp, 'w') as f:
+#         html.render(f)
+#     with open(temp, 'r') as f:
+#         assert f.read() == '<html><body>body content</body><p></p></html>'
+
+
+def prepare(elements):
+    """ Generates the HTML output based on given indents and
+    elements to compare rendered HTML to.
+    Args:
+        elements (list of tuples) : indent, element pairs
+    Returns:
+        str : elements as rendered HTML
+    """
+    output = '\n'.join([
+        '    ' * (indent + 1) + element for indent, element in elements])
+    output = '<!DOCTYPE html>\n{}\n'.format(output)
+    return output
+
+
+def check_render(html, expected):
```

Comment:
good idea to put this in a function, but careful about doing checks for specific complete renders -- not very flexible.

better to have this function return what was rendered, and do the comparison in the tests.